src/TinyHelpers.AspNetCore/Attributes/ContentTypeAttribute.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+using System.Globalization;
+using Microsoft.AspNetCore.Http;
+
+namespace TinyHelpers.AspNetCore.Attributes;
+
+public enum FileType
+{
+    Image,
+    Video,
+    Audio
+}
+
+[AttributeUsage(AttributeTargets.Property | AttributeTargets.Field | AttributeTargets.Parameter, AllowMultiple = false)]
+public class ContentTypeAttribute : ValidationAttribute
+{
+    private readonly IEnumerable<string> validContentTypes;
+
+    private static readonly IEnumerable<string> imageContentTypes = new string[] { "image/jpeg", "image/png", "image/gif", "image/bmp", "image/cis-cod", "image/ief", "image/pipeg", "image/svg+xml", "image/tiff", "image/x-cmu-raster", "image/x-cmx", "image/x-icon", "image/x-portable-anymap", "image/x-portable-bitmap", "image/x-portable-graymap", "image/x-portable-pixmap", "image/x-rgb", "image/x-xbitmap", "image/x-xpixmap", "image/x-xwindowdump" };
+    private static readonly IEnumerable<string> videoContentTypes = new string[] { "video/mp4", "video/ogg", "video/quicktime", "video/mpeg", "video/x-la-asf", "video/x-ms-asf", "video/x-msvideo", "video/x-sgi-movie" };
+    private static readonly IEnumerable<string> audioContentTypes = new string[] { "audio/basic", "audio/mid", "audio/mpeg", "audio/x-wav", "audio/x-mpegurl", "audio/x-pn-realaudio" };
```

Comment:
Il tipo di dato può essere semplificato, non è necessario specificare che l'array è di `string`, perché viene dedotto in automatico dal contenuto.

Reply:
Ho tolto `string` dai Content-Types.

Non l'ho tolto da `validContentTypes` perché richiamo la funzione `contains()` alla linea 48 e quindi ho dovuto fare un cast a livello di switch (non mi piace molto onestamente, se hai altre idee ben vengano)

Reply:
Era il contrario, io mi riferivo all'array di string, quindi quando scrivi `new string[]`, in realtà può essere semplicemente `new []` (e quindi puoi rimettere `IEnuemrable<string>` ed eliminare il cast).